File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head == null || head.next == null) return head;
        ListNode temp = head;
        int size = 1;

        while(temp.next != null){
            temp = temp.next;
            size++;
        }

        temp.next = head;
        int rotation = k % size;
        int skip = size - rotation;
        ListNode last = head;

        for(int i = 1; i < skip; i++){
            last = last.next;
        }
        head = last.next;
        last.next = null;

        return head;
    
        
    }
}

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or not head.next or k == 0:
            return head
        
        temp = head
        size = 1

        while temp.next:
            temp = temp.next
            size += 1

        temp.next = head
        rotation = k % size
        skip = size - rotation
        last = head

        for i in range(1, skip):
            last = last.next

        head = last.next
        last.next = None

        return head

